File: problem-4/prb4.py
x=list(str(input()))  
y=list(str(input()))
z=list(m.upper() for m in str(input()))  #directly converting user input to uppercase and storing it in the list

for i in range(0,len(x)):
    if(x[i]=='a' or x[i]=='e' or x[i]=='i' or x[i]=='o' or x[i]=='u' or x[i]=='A' or x[i]=='E' or x[i]=='I' or x[i]=='O' or x[i]=='U'):
        x[i]='%'  #if character is vowel converting it to '%'
for j in range(0,len(y)):
    if(y[j]=='a' or y[j]=='e' or y[j]=='i' or y[j]=='o' or y[j]=='u' or y[j]=='A' or y[j]=='E' or y[j]=='I' or y[j]=='O' or y[j]=='U'):
        pass
    else:
        y[j]='#'  #if character is consonant converting it to '#'

result=(''.join(x+y+z))
print(result)

#if the output pattern is to display each word separately then replace the last two lines of above code with the following snippet
s1=(''.join(x))
s2=(''.join(y))
s3=(''.join(z))
print(s1, s2, s3)  #this prints all the transformed strings separately

File: problem-4/prb4.c
#include<stdio.h>

int main()
{
    char a[100],b[100],c[100];  //taking 3 words as separate input into 3 different arrays
    scanf("%s",a);
    scanf("%s",b);
    scanf("%s",c);
    int i,j,k;
    for(i=0;a[i]!='\0';i++)
    {
        if(a[i]=='a'||a[i]=='e'||a[i]=='i'||a[i]=='o'||a[i]=='u'||a[i]=='A'||a[i]=='E'||a[i]=='I'||a[i]=='O'||a[i]=='U')
        {
            a[i]='%';  //converting all vowels to % character in the first word
        }
    }
    for(j=0;b[j]!='\0';j++)
    {
        if(!(b[j]=='a'||b[j]=='e'||b[j]=='i'||b[j]=='o'||b[j]=='u'||b[j]=='A'||b[j]=='E'||b[j]=='I'||b[j]=='O'||b[j]=='U'))
        {
            b[j]='#';  //converting all consonants to # character in the second word
        }
    }
    for(k=0;c[k]!='\0';k++)
    {
        if(c[k]>='a'&&c[k]<='z')
        {
            c[k]=c[k]-32;  //converting all characters to uppercase in the third word
        }
    }
    printf("%s%s%s",a,b,c);  //concatenating all transformed strings
    return 0;
}
